mantis-server/mantis-server-worker-client/src/main/java/io/mantisrx/server/worker/client/SseWorkerConnection.java

```diff
@@ -249,18 +249,34 @@ public synchronized Observable<MantisServerSentEvent> call() {
                             return streamContent(response, updateDataRecvngStatus, dataRecvTimeoutSecs, delimiter);
                         })
                         .doOnError((Throwable throwable) -> {
+                            // Only reset connection status, do not close SSE http client.
+                            // otherwise it would cause infinite retry loop by the retryWhen below
                             resetConnected();
-                            logger.warn(getName() +
-                                    "Error on getting response from SSE server: " + throwable.getMessage());
+                            logger.warn("{}: Error on getting response from SSE server: {}",
+                                getName(), throwable.getMessage());
                             connectionResetHandler.call(throwable);
                         })
                         .retryWhen(retryLogic)
-                        .doOnCompleted(this::resetConnected);
+                        .doOnError((Throwable throwable) -> {
```

Comment:
Having double `doOnError` (here and on line 251) seems wrong to me.

Reply:
I think both will still be called, regardless that there is a `retryWhen` in between.

Reply:
let me try test this. My understanding is the retryWhen will swallow the error so outer doOnError won't see it if it's being retried